src/promptflow/promptflow/_utils/multimedia_utils.py

```diff
@@ -2,259 +2,488 @@
-    image_path = (relative_path / file_name).as_posix() if relative_path else file_name
-    if use_absolute_path:
-        image_path = Path(folder_path / image_path).resolve().as_posix()
-    image_reference = {f"data:{image._mime_type};path": image_path}
-    path = folder_path / relative_path if relative_path else folder_path
-    os.makedirs(path, exist_ok=True)
-    with open(os.path.join(path, file_name), "wb") as file:
-        file.write(image)
-    return image_reference
-
-
-def get_file_reference_encoder(folder_path: Path, relative_path: Path = None, *, use_absolute_path=False) -> Callable:
-    def pfbytes_file_reference_encoder(obj):
-        """Dumps PFBytes to a file and returns its reference."""
-        if obj.source_url:
-            return {f"data:{obj._mime_type};url": obj.source_url}
-        if isinstance(obj, PFBytes):
-            file_name = str(uuid.uuid4())
-            # If use_absolute_path is True, the image file path in image dictionary will be absolute path.
-            return _save_image_to_file(obj, file_name, folder_path, relative_path, use_absolute_path)
-        raise TypeError(f"Not supported to dump type '{type(obj).__name__}'.")
-
-    return pfbytes_file_reference_encoder
-
-
-def persist_multimedia_data(value: Any, base_dir: Path, sub_dir: Path = None):
-    pfbytes_file_reference_encoder = get_file_reference_encoder(base_dir, sub_dir)
-    serialization_funcs = {Image: partial(Image.serialize, **{"encoder": pfbytes_file_reference_encoder})}
-    return _process_recursively(value, process_funcs=serialization_funcs)
-
-
-def convert_multimedia_data_to_base64(value: Any, with_type=False, dict_type=False):
-    to_base64_funcs = {PFBytes: partial(PFBytes.to_base64, **{"with_type": with_type, "dict_type": dict_type})}
-    return _process_recursively(value, process_funcs=to_base64_funcs)
-
 
-def convert_multimedia_data_to_string(value: Any, inplace=False):
-    serialization_funcs = {Image: partial(Image.serialize, **{"encoder": None})}
-    return _process_recursively(value, process_funcs=serialization_funcs, inplace=inplace)
-
-
-# TODO: Move this function to a more general place and integrate serialization to this function.
-def _process_recursively(value: Any, process_funcs: Dict[type, Callable] = None, inplace: bool = False) -> dict:
-    if process_funcs:
-        for cls, f in process_funcs.items():
-            if isinstance(value, cls):
-                return f(value)
-    if isinstance(value, list):
-        if inplace:
-            for i in range(len(value)):
-                value[i] = _process_recursively(value[i], process_funcs, inplace)
-        else:
-            return [_process_recursively(v, process_funcs, inplace) for v in value]
-    elif isinstance(value, dict):
-        if inplace:
-            for k, v in value.items():
-                value[k] = _process_recursively(v, process_funcs, inplace)
+    @staticmethod
+    def default_json_encoder(obj):
```

Comment:
We already have a `default_json_encoder` in promptflow._utils.utils, please use that, it should be a general util function, not specified to multimedia.

Reply:
Removed.